config/src/config/network_config.rs

```diff
@@ -342,56 +342,101 @@ impl Default for RateLimitConfig {
 
 pub type TrustedPeerSet = HashMap<PeerId, TrustedPeer>;
 
+// TODO: Combine with RoleType?
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+pub enum TrustedPeerRole {
+    Validator = 0,
+    ValidatorFullNode,
+    FullNode,
+}
```

Comment:
why not define all the roles?

For example, we should have Upstream and PreferredUpstream. The PreferredUpstream should take precedence over everything except Validator. FullNode is an ambiguous role, that should either be Peer or Downstream.

Reply:
I had some more roles, and I stopped at `FullNode` while I thought more about it and how it's integrating with the rest.

Reply:
it seems you're still thinking about the roles, but just wanted to add: the "Trusted" prefix is a bit confusing to me, especially if we start adding non-Validator and non-ValidatorFullNode roles, which are... not trusted.

Reply:
I mean, sure we can remove the `Trusted` name, but the point being is these are roles of known peers.